src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/Thread.NativeAot.cs

```diff
@@ -483,6 +483,41 @@ private static void StopThread(Thread thread)
             }
         }
 
+
+        private static int StackSizeFromConfig
+        {
+            get => _stackSizeFromConfig ??= GetDefaultStackSize();
+        }
+
+        private static int? _stackSizeFromConfig;
+
+        internal static int GetDefaultStackSize()
+        {
+            // Keep the same arbitrary minimum and maximum from the coreclr\vm layer.
+            const uint minStack = 0x10000;     // 64K
+            const uint maxStack = 0x80000000;  //  2G
+
+            const string stackSizeVar = "DefaultStackSize";
+            const string stackSizeEnvVar = "DOTNET_DefaultStackSize";
+
+            // Values in RuntimeHostConfigurationOptions get mapped as knobs, knobs get exposed through
+            // the AppContext class.
+            // To have similar behavior as RhConfig, also read from environment variables, which have priority.
+
+            string? valueFromConfig = Environment.GetEnvironmentVariable(stackSizeEnvVar) ?? AppContext.GetData(stackSizeVar)?.ToString();
```

Comment:
This can use AppContextHelper.GetInt32ComPlusOrDotNetConfig so that we only have one copy of the policy around parsing these.

Reply:
Alternatively, this can call into the native runtime to reuse the value parsed there.